helix-term/src/ui/editor.rs

```diff
@@ -530,19 +530,33 @@ impl EditorView {
                                 (grapheme.as_ref(), width)
                             };
 
+                            let style = if is_whitespace {
+                                style.patch(whitespace_style)
+                            } else {
+                                style
+                            };
+
+                            let cut_off_start = offset.col.saturating_sub(visual_x as usize);
+
                             if !out_of_bounds {
                                 // if we're offscreen just keep going until we hit a new line
                                 surface.set_string(
                                     viewport.x + visual_x - offset.col as u16,
                                     viewport.y + line,
                                     display_grapheme,
-                                    if is_whitespace {
-                                        style.patch(whitespace_style)
-                                    } else {
-                                        style
-                                    },
+                                    style,
                                 );
+                            } else if cut_off_start != 0 && cut_off_start < width {
+                                // partially on screen
+                                let rect = Rect::new(
+                                    viewport.x as u16,
+                                    viewport.y + line,
+                                    (width - cut_off_start) as u16,
+                                    1,
+                                );
```

Comment:
Testing this out with something non-whitespace-like but multi-width like "😀", we render an empty space instead of any character (because of the `rect` here). I think that's a good choice but I wonder if there's anything better to signal that a multi-width character is getting chopped off here?

I would say `…` but then we're rendering something completely different than the actual source. Maybe the plain rect is the best choice? ¯\\\_(ツ)_/¯

Reply:
I considered rendering a character to indicate that a wide grapheme has been cut off, but I think keeping it simple and showing nothing is the right approach.
If we were to render a character it should probably be configurable which char to use, (like how it is with whitespace) but I think that's overkill.